File: android_java_hummus/src/main/java/com/github/klee0kai/hummus/arch/mvp/WeakViewsList.java
package com.github.klee0kai.hummus.arch.mvp;

import com.github.klee0kai.hummus.collections.weaklist.WeakList;
import com.github.klee0kai.hummus.threads.AndroidThreads;

public class WeakViewsList extends WeakList<IRefreshView> {

    public void refreshAllViews() {
        AndroidThreads.runMain(() -> {
            for (IRefreshView v : this)
                v.refreshUI();
        });
    }

    public void refreshAllViews(int delay) {
        AndroidThreads.runMainDelayed(delay, () -> {
            for (IRefreshView v : this)
                v.refreshUI();
        });
    }

}

File: android_kotlin_hummus/src/main/kotlin/com/github/klee0kai/hummus/arch/mvp/WeakViewsList.kt
package com.github.klee0kai.hummus.arch.mvp

import com.github.klee0kai.hummus.collections.weaklist.WeakList
import com.github.klee0kai.hummus.threads.Threads
import com.github.klee0kai.hummus.threads.runMain
import com.github.klee0kai.hummus.threads.runMainDelayed

open class WeakViewsList : WeakList<IRefreshView?>() {

    fun refreshAllViews() {
        Threads.runMain { for (v in this) v?.refreshUI() }
    }

    fun refreshAllViews(delay: Int) {
        Threads.runMainDelayed(delay.toLong()) { for (v in this) v?.refreshUI() }
    }

}